shared_model/interfaces/common_objects/impl/amount.cpp

```diff
@@ -16,23 +17,23 @@ namespace shared_model {
         : amount_(std::move(amount)),
           precision_(0),
           multiprecision_repr_([this] {
-            static const std::regex r("([0-9]+)(\\.([0-9]+))?");
+            static const std::regex r(
+                "[1-9][0-9]*(\\.([0-9]+))?|0\\.([0-9]*[1-9])");
```

Comment:
The same regex pattern is repeated twice - not good, of something will change. Can't you reuse it from `FieldValidator::amount_pattern_`? Or maybe declare the way both classes would see and use it?

Reply:
They're not the same actually. Take in account groups